backend/src/Core/FurnitureShop.Core.Services/CQRS/Mobile/Users/UpdateProfileCH.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Threading.Tasks;
+using FurnitureShop.Core.Contracts.Mobile.Users;
+using FurnitureShop.Core.Domain;
+using FurnitureShop.Core.Services.DataAccess.Entities;
+using LeanCode.DomainModels.DataAccess;
+using Microsoft.AspNetCore.Identity;
+
+namespace FurnitureShop.Core.Services.CQRS.Mobile.Users
+{
+    public class UpdateProfileCH : ICommandHandler<UpdateProfile>
+    {
+        private readonly Serilog.ILogger logger = Serilog.Log.ForContext<UpdateProfileCH>();
+        private readonly UserManager<AuthUser> userManager;
+
+        private readonly IRepository<User> users;
+
+        public UpdateProfileCH(UserManager<AuthUser> userManager, IRepository<User> users)
+        {
+            this.userManager = userManager;
+            this.users = users;
+        }
+
+        public async Task ExecuteAsync(CoreContext context, UpdateProfile command)
+        {
+            var user = await users.FindAndEnsureExistsAsync(context.UserId);
+            var authUser = await userManager.FindByIdAsync(context.UserId.ToString());
+            
+            user.updateUser(
+                command.Firstname,
+                command.Surname,
+                command.Username
+            );
+
+            await users.UpdateAsync(user);
+            // I dont know if here should be updated auth table
```

Comment:
Masz zdanie na to miejsce?

Reply:
na user managerze wywołujesz userManager.UpdateAsync